aip/general/0162.md

```diff
@@ -25,198 +25,169 @@ to the version of an API as a whole.
 
 ## Guidance
 
-APIs **may** store a revision history for a resource if it is useful to users.
+APIs **may** store a revision history for a resource. Examples of when it is
+useful include:
+
+- When it is valuable to expose older versions of a resource via an API. This
+  can avoid the overhead of the customers having to write their own API to store
+  and enable retrieval of revisions.
+- Other resources depend on different revisions of a resource.
+- There is a need to represent the change of a resource over time.
+
+APIs implementing resources with a revision history **should** abstract resource
+revisions as nested collection of the resource. Sometimes, the revisions
+collection can be a top level collection, exceptions include:
+
+- If resource revisions are meant to have longer lifespan than the parent
+resource. In other words, resource revisions exist after resource deletion.
 
-APIs implementing resources with a revision history **must** provide a
-`revision_id` field on the resource:
 
 ```proto
-message Book {
-  // Name and other fields…
+message BookRevision {
+   // The name of the book revision.
+   string name = 1;
 
-  // The revision ID of the book.
-  // A new revision is committed whenever the book is changed in any way.
-  // The format is an 8-character hexadecimal string.
-  string revision_id = 5 [
-    (google.api.field_behavior) = IMMUTABLE,
-    (google.api.field_behavior) = OUTPUT_ONLY];
+   // The snapshot of the book
+   Book snapshot = 2;
 
   // The timestamp that the revision was created.
-  google.protobuf.Timestamp revision_create_time = 6
+  google.protobuf.Timestamp create_time = 3
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Other revision IDs that share the same snapshot.
+  repeated string alternate_ids = 4
```

Comment:
Should we rename this to `aliases`? (That would be more consistent with other changes.)

Reply:
agreed, done.